src/tool/hpcrun/gpu/gpu-binary.c

```diff
@@ -131,3 +151,56 @@ gpu_binary_compute_hash_string
   }
   return used;
 }
+
+uint32_t
+gpu_binary_loadmap_insert
+(
+  const char *device_file,
+  bool mark_used
+)
+{
+  uint32_t loadmap_module_id;
+  load_module_t *module = NULL;
+
+  hpcrun_loadmap_lock();
+  if ((module = hpcrun_loadmap_findByName(device_file)) == NULL) {
+    loadmap_module_id = hpcrun_loadModule_add(device_file);
+    module = hpcrun_loadmap_findById(loadmap_module_id);
+  } else {
+    loadmap_module_id = module->id;
+  }
+  if (mark_used) {
+    hpcrun_loadModule_flags_set(module, LOADMAP_ENTRY_ANALYZE);
+  }
+  hpcrun_loadmap_unlock();
+
+  return loadmap_module_id;
+}
+
+
+bool
+gpu_binary_save
+(
+ const char *mem_ptr,
+ size_t mem_size,
+ bool mark_used,
+ uint32_t *loadmap_module_id
+)
+{
+  // Generate a hash for the binary
+  char *hash_buf = (char*)malloc(HASH_LENGTH * 2);
```

Comment:
`hash_buf` is allocated but never freed (AFAICT).

Reply:
replaced with stack variable to avoid leak.